### PR TITLE
Controller view implementation

### PR DESCRIPTION
**Pull request by TheSlimvReal**
I implemented most of the logic for the view and controller.

If the models are finished, you need to implement the execute and validate mehtod for the specific command subclass.

### CODE REVIEW

File: modules/controller/command_parser.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from modules.controller.commands.label_command import LabelCommand
+from modules.controller.commands.quit_command import QuitCommand
+from modules.controller.commands.train_command import TrainCommand
+from modules.exception.excpetions import IllegalArgumentException
+
+
+class CommandParser:
+
+    _valid_commands: Dict[str, Command.__class__] = {
+        "collect": CollectorCommand,
```
Comment (annaric) [inline modules/controller/command_parser.py]: why is the naming for the command at the one hand a substantive (labeler) but on the other hand a verb (collect)?
Isn't that inconsistent?

File: modules/controller/command_parser.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "classify": ClassifyCommand,
+        "quit": QuitCommand,
+    }
+
+    ##  Parses a string to a command
+    #
+    #   @param input_string the string that should be parsed to a command
+    @staticmethod
+    def parse_input(input_string: str) -> Command:
+        arg_list = input_string.split(" ")
```
Comment (annaric) [inline modules/controller/command_parser.py]: How is the input of the variable? Aren't there commas in between the different input vairables? Or is it always just a blank?

File: modules/controller/command_parser.py (hunk 1 above)
Comment (TheSlimvReal) [reply modules/controller/command_parser.py]: I changed it in the new commit, please review

**Comment by yannickfunk** [comment]
please use double underscores "__" for private attributes and the @property tag for getters and @attr.setter tag for setters as you already did

**Comment by yannickfunk** [comment]
Just to be consistent, I think you partially mixed up interfaces and abstract classes because your Command class contains some logic, but is an interfaces in our design. In addition to this we should avoid to declare attributes before the constructor and out of methods, because this sets the attributes to class attributes (static attributes)

**Comment by TheSlimvReal** [comment]
> 
> 
> Just to be consistent, I think you partially mixed up interfaces and abstract classes because your Command class contains some logic, but is an interfaces in our design. In addition to this we should avoid to declare attributes before the constructor and out of methods, because this sets the attributes to class attributes (static attributes)

I know that I originally planned the Command class to be an interface, but during implementation, it turned out to make way more sense in designing it as an abstract class, due to a lot of logic that can be shared this way.

**Pull request merged**